### ISSUE TITLE
Fix Style/MethodDefParentheses issue with annonymous block forwarding

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#10321]
## Expected behavior

I would expect `Style/MethodDefParentheses` to not throw errors when anonymous block forwarding is used because anonymous block forwarding appears to require the use of parenthesis while explicit `&block` usage does not.

## Actual behavior

When enabling the `Naming/BlockForwarding` cop and correcting the code accordingly, you'll start getting errors with `Style/MethodDefParentheses`.

## Steps to reproduce the problem

Start by configuring Rubocop to use the following configuration:

``` 
Naming/BlockForwarding:
  Enabled: true
Style/MethodDefParentheses:
  EnforcedStyle: require_no_parentheses
```

Next, use the following code as an example:

``` ruby
def fetch_value key, *default_value, &block
  fetch(key, *default_value, &block) || default_value.first
end
```

Save the above and run as `rubocop snippet.rb`. Notice you get `Naming/BlockForwarding` errors, now change the above implementation to use anonymous block forwarding as shown below:

``` ruby
def fetch_value(key, *default_value, &)
  fetch(key, *default_value, &) || default_value.first
end
```

Notice you now have a single `Style/MethodDefParentheses` error now.

## Notes

I'd like to note that this behavior is similar to what was reported [in this earlier issue](https://github.com/rubocop/rubocop/issues/10317). Also, it's odd that Ruby 3.1.0 requires the use of parenthesis when using anonymous block forwarding. Is this a possible Ruby 3.1.0 bug? I tried searching Ruby Issues but didn't see anything that hinted at this new behavior.

## RuboCop version

```
1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.0, running on ruby 3.1.0 arm64-darwin21.2.0)
  - rubocop-performance 1.13.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.7.0
```

**Comment by koic** [comment]
Good catch! Method definitions using anonymous block forwarding require argument parentheses. AFAIK, this issue is a known and expected behavior in Ruby 3.1. First adjust `Style/MethodDefParentheses` cop by #10323, and I will continue to make several adjustments to other cops for the syntax.

It's up to you to send feedback to the Ruby core 😃

**Comment by bkuhlmann** [comment]
🙇 Koichi. I piggybacked onto your punning issue and logged a [new issue](https://bugs.ruby-lang.org/issues/18441). This might not go anywhere but wanted to ask.
